backend/controllers/appointmentConfirmDeletion.js

```diff
@@ -0,0 +1,86 @@
+const packAppointmentDetails = (req, details) => {
+  req.flash('appointmentDetails', details)
```

Comment:
I might be missing something but why do we need to store the details? Don't we just need the appointment id?

Reply:
The point is that the appointment details page makes lots of calls and does lots of calculations - so why do it all again here? Also, this is just a confirmation page - it shouldn't be accessible without going to the initial details page first, hence we don't even have the id in the route.

Reply:
It's to keep the code simpler. It's acceptable to make a request each time for the appointment details. It's a standard flow when it comes to web.

Reply:
I did this because of the reasons given above and - more importantly - it has already been done for appointments (see the add appointment journey).

Personally I wouldn't want to change it for 2 reasons (if i understand what you are proposing correctly): 

- Code duplication is not good for future maintenance, especially when it is exactly the same code, as this would be w.r.t. the appointmentDetails page - but we could get round this by using a common component/class. Or would that make it too complicated?
- Inefficiency. Why get the same data twice?

It makes it difficult to know what the right thing to do is - there is no documentation for these standard practices AND code already exists that doesn't use these standard practices.

Reply:
We will make a service to provide the appointment info, and change appt details and this page to use that. The flash will be removed.